src/Castle.Windsor/Compatibility/PermissionUtil.cs

```diff
@@ -12,20 +12,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-using System;
+#if FEATURE_SECURITY_PERMISSIONS
 
-#if NET45
-namespace Castle.Core.Internal
+namespace Castle.Windsor.Compatibility
```

Comment:
This doesn't belong in the `Castle.Windsor.Compatibility` namespace which is now empty? Windsor now owns this class and can refactor it however it likes, it isn't a compatibility layer.

It also looks to only be used inside the `Castle.Windsor` assembly so it shouldn't be public.

Reply:
Ok, I'll shift it to the `.Core.Internal` namespace. Given Castle.Core has dropped CAS support, I suspect something similar will happen with Windsor (i.e. this class won't live long), but need to understand it a bit more first.

Reply:
There's no `InternalsVisibleTo` attribute on Castle.Windsor, so if I mark the class `internal` I'll have to get rid of the test. Happy to go either way, although I assume because the attribute isn't there already, the preference is to just not have a test?

Reply:
> Ok, I'll shift it to the `.Core.Internal` namespace. Given Castle.Core has dropped CAS support, I suspect something similar will happen with Windsor (i.e. this class won't live long), but need to understand it a bit more first.

Sounds good.

> There's no `InternalsVisibleTo` attribute on Castle.Windsor, so if I mark the class `internal` I'll have to get rid of the test. Happy to go either way, although I assume because the attribute isn't there already, the preference is to just not have a test?

Yer, we wouldn't want `InternalsVisibleTo`, the unit tests should only test the public API. I don't mind whether you stick it in the Internal namespace and mark it public, or delete the test. I'll let you decide.

Reply:
Haven't had a chance to get back into the code for this, but did some reading up on internals and testing. Background: I'm used to having internals visible to test projects (and this is [what Castle.Core does too](https://github.com/castleproject/Core/blob/v5.0.0/src/Castle.Core/Properties/InternalsVisibleToTests.cs)), as otherwise you can end up with an awful lot of internal code that can't easily be tested independently (as it's hidden under other functionality); wanted to check my reasoning.

It's a bit long-winded, but [LosTechies has a good post about it](https://lostechies.com/derickbailey/2014/01/03/semantics-modules-and-testing-why-and-how-i-test-internal-components-not-private-methods/) that aligns with my thoughts. I quite liked the car engine analogy:
> Engines are not “private” components, in my experience. They are “internal” components of the car. They are complex, prone to faults from many parts, and must be rigorously tested prior to mounting them in the vehicle. Failure to do this leads to cars that have faulty engines.

Having said all that... This is one tiny test for some code that will hopefully get deprecated soon, so not really worth worrying about. I'll just drop the test!

Reply:
Per discussion in #612, this is good to go now - I ended up getting rid of the tests.